<?xml version="1.0"?>
<robot xmlns:xacro="http://www.ros.org/wiki/xacro">
    <xacro:include filename="$(find two_wheels_core)/urdf/materials.xacro"/>
    <link name="base_link">
        <inertial>
            <origin xyz="7.136447167432144e-05 3.453515765351803e-05 0.024408245184796683" rpy="0 0 0"/>
            <mass value="1.000593039456424"/>
            <inertia ixx="0.003708" iyy="0.007317" izz="0.010943" ixy="-0.0" iyz="-0.0" ixz="-0.0"/>
        </inertial>
        <visual>
            <origin xyz="0 0 0" rpy="0 0 0"/>
            <geometry>
                <mesh filename="file://$(find two_wheels_core)/urdf/meshes/base_link.stl" scale="0.001 0.001 0.001"/>
            </geometry>
            <material name="silver"/>
        </visual>
        <collision>
            <origin xyz="0 0 0" rpy="0 0 0"/>
            <geometry>
                <mesh filename="file://$(find two_wheels_core)/urdf/meshes/base_link.stl" scale="0.001 0.001 0.001"/>
            </geometry>
        </collision>
    </link>

    <link name="front_left_wheel">
        <inertial>
            <origin xyz="1.625960121487413e-06 -0.03356595189466297 -1.9609319341917786e-09" rpy="0 0 0"/>
            <mass value="0.21820489062799583"/>
            <inertia ixx="3.8e-05" iyy="4.8e-05" izz="3.8e-05" ixy="0.0" iyz="0.0" ixz="0.0"/>
        </inertial>
        <visual>
            <origin xyz="0.077933 0.059968 -0.023743" rpy="0 0 0"/>
            <geometry>
                <mesh filename="file://$(find two_wheels_core)/urdf/meshes/front_left_wheel.stl"
                      scale="0.001 0.001 0.001"/>
            </geometry>
            <material name="silver"/>
        </visual>
        <collision>
            <origin xyz="0.077933 0.059968 -0.023743" rpy="0 0 0"/>
            <geometry>
                <mesh filename="file://$(find two_wheels_core)/urdf/meshes/front_left_wheel.stl"
                      scale="0.001 0.001 0.001"/>
            </geometry>
        </collision>
    </link>

    <link name="rear_left_wheel">
        <inertial>
            <origin xyz="1.3407781088436987e-06 -0.03356595189466117 -1.9609318023527944e-09" rpy="0 0 0"/>
            <mass value="0.2182048906279955"/>
            <inertia ixx="3.8e-05" iyy="4.8e-05" izz="3.8e-05" ixy="0.0" iyz="0.0" ixz="0.0"/>
        </inertial>
        <visual>
            <origin xyz="-0.077933 0.059968 -0.023743" rpy="0 0 0"/>
            <geometry>
                <mesh filename="file://$(find two_wheels_core)/urdf/meshes/rear_left_wheel.stl"
                      scale="0.001 0.001 0.001"/>
            </geometry>
            <material name="silver"/>
        </visual>
        <collision>
            <origin xyz="-0.077933 0.059968 -0.023743" rpy="0 0 0"/>
            <geometry>
                <mesh filename="file://$(find two_wheels_core)/urdf/meshes/rear_left_wheel.stl"
                      scale="0.001 0.001 0.001"/>
            </geometry>
        </collision>
    </link>

    <link name="rear_right_wheel">
        <inertial>
            <origin xyz="1.4899030495729715e-06 0.03356602362407317 3.1974513525767634e-07" rpy="0 0 0"/>
            <mass value="0.2182048906279955"/>
            <inertia ixx="3.8e-05" iyy="4.8e-05" izz="3.8e-05" ixy="-0.0" iyz="0.0" ixz="-0.0"/>
        </inertial>
        <visual>
            <origin xyz="-0.077931 -0.060032 -0.023742" rpy="0 0 0"/>
            <geometry>
                <mesh filename="file://$(find two_wheels_core)/urdf/meshes/rear_right_wheel.stl"
                      scale="0.001 0.001 0.001"/>
            </geometry>
            <material name="silver"/>
        </visual>
        <collision>
            <origin xyz="-0.077931 -0.060032 -0.023742" rpy="0 0 0"/>
            <geometry>
                <mesh filename="file://$(find two_wheels_core)/urdf/meshes/rear_right_wheel.stl"
                      scale="0.001 0.001 0.001"/>
            </geometry>
        </collision>
    </link>

    <link name="front_right_wheel">
        <inertial>
            <origin xyz="-1.489903047255381e-06 0.033566023624073626 8.554326641832533e-08" rpy="0 0 0"/>
            <mass value="0.2182048906279955"/>
            <inertia ixx="3.8e-05" iyy="4.8e-05" izz="3.8e-05" ixy="0.0" iyz="-0.0" ixz="-0.0"/>
        </inertial>
        <visual>
            <origin xyz="0.077931 -0.060032 -0.023742" rpy="0 0 0"/>
            <geometry>
                <mesh filename="file://$(find two_wheels_core)/urdf/meshes/front_right_wheel.stl"
                      scale="0.001 0.001 0.001"/>
            </geometry>
            <material name="silver"/>
        </visual>
        <collision>
            <origin xyz="0.077931 -0.060032 -0.023742" rpy="0 0 0"/>
            <geometry>
                <mesh filename="file://$(find two_wheels_core)/urdf/meshes/front_right_wheel.stl"
                      scale="0.001 0.001 0.001"/>
            </geometry>
        </collision>
    </link>

    <joint name="front_left_wheel_joint" type="continuous">
        <origin xyz="-0.077933 -0.059968 0.023743" rpy="0 0 0"/>
        <parent link="base_link"/>
        <child link="front_left_wheel"/>
        <axis xyz="0.0 -1.0 0.0"/>
    </joint>

    <joint name="rear_left_wheel_joint" type="continuous">
        <origin xyz="0.077933 -0.059968 0.023743" rpy="0 0 0"/>
        <parent link="base_link"/>
        <child link="rear_left_wheel"/>
        <axis xyz="0.0 -1.0 0.0"/>
    </joint>

    <joint name="rear_right_wheel_joint" type="continuous">
        <origin xyz="0.077931 0.060032 0.023742" rpy="0 0 0"/>
        <parent link="base_link"/>
        <child link="rear_right_wheel"/>
        <axis xyz="-0.0 1.0 -0.0"/>
    </joint>

    <joint name="front_right_wheel_joint" type="continuous">
        <origin xyz="-0.077931 0.060032 0.023742" rpy="0 0 0"/>
        <parent link="base_link"/>
        <child link="front_right_wheel"/>
        <axis xyz="0.0 1.0 -0.0"/>
    </joint>

    <link name="base_footprint"/>
    <joint name="base_footprint_joint" type="fixed">
        <parent link="base_link"/>
        <child link="base_footprint"/>
        <origin xyz="0 0 0" rpy="0 0 0"/>
    </joint>

</robot>